src/event-handler.js

```diff
@@ -1,37 +1,93 @@
-var optimizelyEvents = require('./optimizely-defined-events');
+var optimizelyWebXEvents = require('./optimizely-x-defined-events');
+var optimizelyFullStackEvents = require('./optimizely-fs-defined-events');
 
 function EventHandler(common) {
     this.common = common || {};
 }
 
 EventHandler.prototype.logEvent = function(event) {
-    if (optimizelyEvents.events[event.EventName]) {
-        var optimizelyEvent = {
+    if (optimizelyWebXEvents.events[event.EventName]) {
+        var optimizelyWebXEvent = {
             type: 'event',
             eventName: event.EventName
         };
 
         if (event.EventAttributes) {
-            optimizelyEvent.tags = event.EventAttributes;
+            optimizelyWebXEvent.tags = event.EventAttributes;
         }
 
         if (event.CustomFlags && event.CustomFlags['Optimizely.Value']) {
-            optimizelyEvent.tags.value = event.CustomFlags['Optimizely.Value'];
+            optimizelyWebXEvent.tags.value = event.CustomFlags['Optimizely.Value'];
         }
-        window['optimizely'].push(optimizelyEvent);
+        window['optimizely'].push(optimizelyWebXEvent);
+    }
+
+    // if optimizely full stack is being used
+    if (window.optimizelyClientInstance) {
+        var self = this;
+        var eventKey = event.EventName,
+            userId,
+            userAttributes = self.common.userAttributes,
+            eventTags = {};
+
+        if (window.mParticle && window.mParticle.Identity) {
+            var userIdentities = window.mParticle.Identity.getCurrentUser().getUserIdentities()['userIdentites'];
+
+            switch(self.common.userIdField) {
+                case 'customerId':
```

Comment:
this and the other switch statement are missing `deviceApplicationStamp` as an option, so add a 

```
case 'deviceApplicationStamp'
   userId = window.mParticle.getDeviceId();
```